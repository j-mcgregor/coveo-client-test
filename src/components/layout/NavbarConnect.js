import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import {
  fetchResultsFromNav,
  selectProduct,
  updateSuggested
} from '../../flux/actions/queryActions';
import Navbar from './Navbar';

const mapStateToProps = state => ({
  results: state.results,
  errors: state.errors
});

const mapDispatchToProps = dispatch => ({
  fetchResultsFromNav: query => dispatch(fetchResultsFromNav(query)),
  selectProduct: product => dispatch(selectProduct(product)),
  updateSuggested: results => dispatch(updateSuggested(results))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Navbar));
