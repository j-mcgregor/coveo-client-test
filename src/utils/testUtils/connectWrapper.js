import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

export default (initialState, component) => {
  const mockStore = configureStore([thunk]);
  const store = mockStore(initialState);
  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>{component}</BrowserRouter>
    </Provider>
  );

  return wrapper;
};
