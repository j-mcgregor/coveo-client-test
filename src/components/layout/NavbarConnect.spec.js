import React from 'react';
import NavbarConnect from './NavbarConnect';
import connectWrapper from '../../utils/testUtils/connectWrapper';

describe('NavbarConnect', () => {
  it('should render with the store', () => {
    const initialState = {
      errors: {},
      results: {}
    };

    const wrapper = connectWrapper(initialState, <NavbarConnect />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should pass the products to the Navbar', () => {
    const initialState = {
      errors: {},
      results: {
        navResults: ['test']
      }
    };

    const wrapper = connectWrapper(initialState, <NavbarConnect />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Navbar').prop('results')).toEqual({
      navResults: ['test']
    });
  });
});
