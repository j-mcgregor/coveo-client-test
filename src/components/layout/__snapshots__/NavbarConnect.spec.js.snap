// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarConnect should pass the products to the Navbar 1`] = `ReactWrapper {}`;

exports[`NavbarConnect should render with the store 1`] = `ReactWrapper {}`;
